### DISCUSSION TITLE
Check if there are related records before deleting

### DISCUSSION THREAD

**Discussion by nikosid** [filamentphp/filament#6154] [General]
How to check if there are related records before deleting to show user friendly error but not raw mysql error about Integrity constraint violation?
And I want to check existing related records also for situation with Soft Deleting - I don't want to even soft delete record if there are related records.

**Comment by nikosid** [comment]
I found some solution here https://filamentphp.com/tricks/how-to-prevent-delete-record-when-have-a-child-record
But I'm still looking for the solution where I will not hide delete button but show an error message

**Reply by ihahachi** [reply]
I also want a solution to this problem, but I did not find anything other than this solution

**Reply by emd-che** [reply]
I'm interested on a solution too

**Comment by manbhatia** [comment]
I was also looking for this and found that it can be solved by using this:

https://filamentphp.com/docs/2.x/admin/resources/deleting-records#halting-the-deletion-process

Here is the solution that I am using in my project:

https://filamentphp.com/tricks/how-to-check-if-there-are-related-records-before-deleting-to-show-user-friendly-error (I wrote a trick in Filament community)

I hope this helps.

**Reply by nikosid** [reply]
Great! Thank you. This solution is not perfect but almost!
